### ISSUE TITLE
Class constructor ObjectId must be invoked with 'new'

### ISSUE THREAD

**Issue by larhou** [mdn/content#27196]
### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Displaying_data/Genre_detail_page

### What specific section or headline is this issue about?

Note: You might get an error similar to this:

### What information was incorrect, unhelpful, or incomplete?

When I tested the code block in the section "Note: You might get an error similar to this:" I got this error message in the browser:

**Class constructor ObjectId cannot be invoked without 'new'
TypeError: Class constructor ObjectId cannot be invoked without 'new'**

I changed this code:

```
exports.genre_detail = asyncHandler(async (req, res, next) => {
  const id = mongoose.Types.ObjectId(req.params.id);
  // …
});

```
into this code ( so ObjectId is invoked with `new` ) and now it works :

```
exports.genre_detail = asyncHandler(async (req, res, next) => {
  const id = **new** mongoose.Types.ObjectId(req.params.id);
  // …
});

```

### What did you expect to see?

exports.genre_detail = asyncHandler(async (req, res, next) => {
  const id = **new** mongoose.Types.ObjectId(req.params.id);
  // …
});

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/learn/server-side/express_nodejs/displaying_data/genre_detail_page`
* MDN URL: https://developer.mozilla.org/en-US/docs/Learn/Server-side/Express_Nodejs/Displaying_data/Genre_detail_page
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/learn/server-side/express_nodejs/displaying_data/genre_detail_page/index.md
* Last commit: https://github.com/mdn/content/commit/0586fbecaf6683f5e793cd9510397b630fb43e61
* Document last modified: 2023-04-14T05:20:55.000Z

</details>

**Comment by hamishwillee** [comment]
Thank you @larhou 

You're absolutely right about using `new`. However I have fixed this in a different way in https://github.com/mdn/content/pull/27399 .

If you get this error it means that the system is attempting to pass a non-castable ID, which should not happen. So showing people how to manually cast something that is probably an invalid ID is not going to help. Instead I have explained how to test the ID, and pointed to the likely cause - the wrong router being called so your ID is invalid.

**Comment by larhou** [comment]
Thanks Hamish !!Great extra explanation on the site 👍LarsSent from my iPhoneDen 19. jun. 2023 kl. 05.49 skrev Hamish Willee ***@***.***>:﻿
Thank you @larhou
You're absolutely right about using new. However I have fixed this in a different way in #27399 .
If you get this error it means that the system is attempting to pass a non-castable ID, which should not happen. So showing people how to manually cast something that is probably an invalid ID is not going to help. Instead I have explained how to test the ID, and pointed to the likely cause - the wrong router being called so your ID is invalid.
